src/main/java/com/google/cloud/teleport/splunk/SplunkEventWriter.java

```diff
@@ -106,6 +117,7 @@ public static Builder newBuilder() {
   public void setup() {
 
     checkArgument(url().isAccessible(), "url is required for writing events.");
+    checkArgument(isValidUrlFormat(url().get()), INVALID_URL_FORMAT_MESSAGE);
```

Comment:
ReadMessages is a composite transform where messages are read via the built-in PubsubIO [1]. So the messages will still be read by PubsubIO and would not necessarily "fail-fast". I.e the pipeline will be running and this would fail at runtime as we try to publish messages to Splunk.


[1] https://github.com/GoogleCloudPlatform/DataflowTemplates/blob/604b79630b5dbfa61b10e2e0da7dfff7fb35a96f/src/main/java/com/google/cloud/teleport/templates/PubSubToSplunk.java#L326-L328

Reply:
Got it. Thanks a ton :)